scipy/interpolate/_fitpack_repro.py

```diff
@@ -501,12 +575,85 @@ def __call__(self, p):
         _dierckx.qr_reduce(AB, offset, nc, QY, startrow=nc)
 
         # solve for the coefficients
-        c = _dierckx.fpback(AB, nc, QY)
+        c, _, fp = _dierckx.fpback(AB, nc, self.x, self.y, self.t, self.k, self.w, QY)
 
         spl = BSpline(self.t, c, self.k)
-        residuals = _compute_residuals(self.w**2, spl(self.x), self.y)
+        self.spl = spl   # store it
+
+        return fp - self.s
+
+# Solver class for f(p) for periodic splines
+class Fperiodic:
+    def __init__(self, x, y, t, k, s, w=None, *,
+                 R=None, Y=None, A1=None, A2=None, Z=None):
+        self.x = x
+        self.y = y
+        self.t = t
+        self.k = k
+        w = np.ones_like(x, dtype=float) if w is None else w
+        if w.ndim != 1:
+            raise ValueError(f"{w.ndim = } != 1.")
+        self.w = w
+        self.s = s
+
+        if y.ndim != 2:
+            raise ValueError(f"F: expected y.ndim == 2, got {y.ndim = } instead.")
+
+        # ### precompute what we can ###
+
+        # https://github.com/scipy/scipy/blob/maintenance/1.11.x/scipy/interpolate/fitpack/fpcurf.f#L250
+        # c  evaluate the discontinuity jump of the kth derivative of the
+        # c  b-splines at the knots t(l),l=k+2,...n-k-1 and store in b.
+        b, b_offset, b_nc = disc(t, k)
+
+        # the QR factorization of the data matrix, if not provided
+        # NB: otherwise, must be consistent with x,y & s, but this is not checked
+        if ((A1 is None or A2 is None or Z is None) or
+            (R is None and Y is None)):
+            # Ref: https://github.com/scipy/scipy/blob/596b586e25e34bd842b575bac134b4d6924c6556/scipy/interpolate/fitpack/fpperi.f#L171-L215
+            # The computation in the above link is performed for a
+            # given set of knots i.e., t vector
+            (R, A1, A2, Z), _, _, _, _ = _lsq_solve_qr(
+                x, y, t, k, w, periodic=True, solve_for_p=True)
```

Comment:
This is extremely confusing. This seems to be the only place where you use `solve_for_p=True`, but you do not return it.